File: nim/src/dependency/fibonacci.nim
type
    Fibonacci* = ref object

proc new_fibonacci*(): Fibonacci =
    return Fibonacci()

proc compute*(self: Fibonacci, num: uint32): uint64 =
    if num <= 0:
        return 0'u64

    if num == 1:
        return 1'u64

    var fibonacci = @[0'u64,1'u64]
    for i in 2..num:
        try:
            let sum = fibonacci[i - 1] + fibonacci[i - 2]
            fibonacci.add(sum)

        except Exception as ex:
            echo ex.getStackTrace()
            fibonacci.add(0)

    return fibonacci[^1]

File: python/src/dependency/fibonacci.py
# internal
from dependency.math_calculation import MathCalculation


class Fibonacci(MathCalculation):
    
    def __init__(self):
        pass

    def compute(self, num: int) -> int:
        if not isinstance(num, int):
            return -1
        
        if num < 0:
            return 0
        
        if num == 1:
            return 1
        
        fibonacci: list = [0,1]
        for i in range(2,num):
            sum = fibonacci[i - 1] + fibonacci[i - 2]
            fibonacci.append(sum)
        
        return fibonacci[-1]